languages/python/sqlalchemy-oso/sqlalchemy_oso/auth.py

```diff
@@ -79,13 +80,16 @@ def get_field_type(model, field):
 
     combined_filter = None
     has_result = False
+    has_role_allows = False
     for result in results:
         has_result = True
 
         resource_partial = result["bindings"]["resource"]
-        filter = partial_to_filter(
+        filter, role_allows = partial_to_filter(
             resource_partial, session, model, get_model=oso.get_class
         )
+        has_role_allows |= role_allows
```

Comment:
I don't think this logic is correct. If I write a policy like:

```
allow(actor, action, resource) if
    OsoRoles.role_allow(actor, action, resource) or
    resource.public = true;
```

The `role_allow` filter will be applied to both branches.

Reply:
I think if we use a subquery we can apply the role filter only on the branch that is needed.  Generate something like:

```
SELECT * FROM (... roles query) WHERE ... normal filters ...;
```

or maybe

```
with roles_query AS roles_resources
SELECT * FROM organization INNER JOIN roles_resources USING (id) WHERE ... filters
```

Reply:
nvm won't quite work... anyway with more thought I think there's a way to write this that is correct